opm/simulators/flow/GenericOutputBlackoilModule.cpp

```diff
@@ -391,6 +391,54 @@ outputFipAndResvLog(const Inplace& inplace,
     }
 }
 
+template<class FluidSystem>
+void GenericOutputBlackoilModule<FluidSystem>::
+accumulateRftDataParallel(const Parallel::Communication& comm) {
+    if (comm.size() > 1) {
+        collectRftMapOnRoot(oilConnectionPressures_, comm);
+        collectRftMapOnRoot(waterConnectionSaturations_, comm);
+        collectRftMapOnRoot(gasConnectionSaturations_, comm);
+    }
+}
+
+template<class FluidSystem>
+void GenericOutputBlackoilModule<FluidSystem>::
+collectRftMapOnRoot(std::map<std::size_t, Scalar>& local_map, const Parallel::Communication& comm) {
+
+    const auto mapsize = local_map.size();
+    std::vector<int> keys;
+    std::vector<Scalar> values;
```

Comment:
Dune can magically communicate pairs, too. Hence we could use 
```c++
std::vector<std::pair<int,Scalar>> pairs(local_map.begin(), local_map.end());
```
and pass that to `gatherv`

Reply:
Nice, thanks!